packages/core/src/app/checkout/hooks/useLoadCheckout.ts

```diff
@@ -43,12 +44,24 @@ export const useLoadCheckout = (checkoutId: string): {isLoadingCheckout: boolean
     };
 
     useEffect(() => {
-        fetchDataWithRetry()
-            .then(() => setIsLoadingCheckout(false))
-            .catch((error) => {
-                throw error;
+        if (!isLoadingCheckout) {
+            return;
+        }
+
+        if (!initialState) {
+            // If the initial data has not been preloaded from the server, we need to make API calls to fetch it.
+            fetchDataWithRetry()
+                .then(() => setIsLoadingCheckout(false))
+                .catch((error) => {
+                    throw error;
+                });
+        } else {
+            requestAnimationFrame(async () => {
```

Comment:
Is there a reason why we need to wrap the initial render and hydration calls in `requestAnimationFrame`?

Reply:
The idea is to spread the work across multiple tasks so we don't do too much in a single one, as this could negatively impact Total Blocking Time. Having said that, I notice I'm using two, so I'll look into whether I need both.